src/include/distributed/citus_nodes.h

```diff
@@ -92,38 +92,21 @@ CitusNodeTagI(Node *node)
 	return ((CitusNode*)(node))->citus_tag;
 }
 
-/*
- * Postgres's nodes/nodes.h has more information on why we do this.
- */
-#ifdef __GNUC__
 
 /* Citus variant of newNode(), don't use directly. */
-#define CitusNewNode(size, tag) \
-({	CitusNode   *_result; \
-	AssertMacro((size) >= sizeof(CitusNode));		/* need the tag, at least */ \
-	_result = (CitusNode *) palloc0fast(size); \
-	_result->extensible.type = T_ExtensibleNode; \
-	_result->extensible.extnodename = CitusNodeTagNames[tag - CITUS_NODE_TAG_START]; \
-	_result->citus_tag =(int) (tag); \
-	_result; \
-})
-
-#else
-
-extern CitusNode *newCitusNodeMacroHolder;
-
-#define CitusNewNode(size, tag) \
-( \
-	AssertMacro((size) >= sizeof(CitusNode)),		/* need the tag, at least */ \
-	newCitusNodeMacroHolder = (CitusNode *) palloc0fast(size), \
-	newCitusNodeMacroHolder->extensible.type = T_ExtensibleNode, \
-	newCitusNodeMacroHolder->extensible.extnodename = CitusNodeTagNames[tag - CITUS_NODE_TAG_START], \
-	newCitusNodeMacroHolder->citus_tag =(int) (tag), \
-	newCitusNodeMacroHolder \
-)
-
-#endif
+static inline CitusNode *
```

Comment:
Should we delete  two lines here:
https://github.com/citusdata/citus/blob/f6959715dc49404510f2f3340b9719b33ddd152b/src/backend/distributed/utils/citus_nodefuncs.c#L56

Reply:
We still have CitusNewNode(size_t size, CitusNodeTag tag) function, so I guess not.

Reply:
`CitusNode *newCitusNodeMacroHolder;` is not referenced anywhere. So I assumed it was related to getting around an issue with macro usage. Now since we have a function instead, this line may be redundant and confusing.

Reply:
Just noticed that, you are right. It is redundant. Let me remove it.